invenio_rdm_migrator/streams/actions/load/communities.py

```diff
@@ -0,0 +1,242 @@
+            yield Operation(OperationType.INSERT, FilesInstance, file_instance)
+
+            # update the old object version also if there was one
+            if logo_object_version_id:
+                yield Operation(
+                    OperationType.UPDATE,
+                    FilesObjectVersion,
+                    {"version_id": logo_object_version_id, "is_head": False},
+                )
+
+            object_version = self.data.object_version
+            logo_object_version_id = object_version["version_id"]
+            object_version["bucket_id"] = bucket_id
+            yield Operation(OperationType.INSERT, FilesObjectVersion, object_version)
+
+            community_file = self.data.community_file
+            community_file["record_id"] = community_id
+            community_file["object_version_id"] = logo_object_version_id
+
+            if community_file_id:
+                community_file["id"] = community_file_id
+                yield Operation(OperationType.UPDATE, CommunityFile, community_file)
+            else:  # no logo before, so we do an INSERT
+                community_file_id = community_file["id"]
+                yield Operation(OperationType.INSERT, CommunityFile, community_file)
+
+        # Update the state
+        STATE.COMMUNITIES.update(
+            slug,
+            {
+                "id": community_id,
+                "bucket_id": bucket_id,
+                "community_file_id": community_file_id,
+                "logo_object_version_id": logo_object_version_id,
+            },
+        )
+
+
+@dataclass
+class CommunityDeleteData(LoadData):
+    """Community action data."""
+
+    community: dict
+
+
+class CommunityDeleteAction(LoadAction):
+    """Delete a community."""
+
+    name = "community-delete"
+    data_cls = CommunityDeleteData
+
+    def _generate_rows(self, **kwargs):
+        """Generates rows for deleting a community."""
+        slug = self.data.community["slug"]
+        state = STATE.COMMUNITIES.get(slug)
+        community_id = state["id"]
+        oai_set_id = state["oai_set_id"]
+        community_file_id = state["community_file_id"]
+        logo_object_version_id = state["logo_object_version_id"]
+
+        # We only soft-delete by setting `json = NULL`
+        yield Operation(
+            OperationType.UPDATE,
+            Community,
+            {"id": community_id, "json": None},
+        )
+
+        # We hard-delete the OAISet
+        yield Operation(OperationType.DELETE, OAISet, {"id": oai_set_id})
+
+        # Clean-up the logo also if there is one
+        if community_file_id:
+            yield Operation(
+                OperationType.DELETE,
+                FilesObjectVersion,
+                {"version_id": logo_object_version_id},
+            )
+            yield Operation(
+                OperationType.DELETE, CommunityFile, {"id": community_file_id}
+            )
+
+        # NOTE: We don't update the state actually
```

Comment:
why?

Reply:
Mainly because there's no `.delete()` method. But on the other hand, I'm not sure if we actually want to fully delete rows from the state... Maybe we could add a `deleted_at` column and keep the Tx ID or something similar?

Reply:
ahh yes, I've added `delete` and `search`